message.go

```diff
@@ -168,13 +167,42 @@ type MessageReactions struct {
 
 // ContentWithMentionsReplaced will replace all @<id> mentions with the
 // username of the mention.
-func (m *Message) ContentWithMentionsReplaced() string {
-	if m.Mentions == nil {
-		return m.Content
+func (m *Message) ContentWithMentionsReplaced(s *Session) (content string, err error) {
```

Comment:
This is breaking the API, we should have a ContentWithMentionsReplacedWithSession and document that this will replace more data than above.